kore/src/Kore/Log.hs

```diff
@@ -208,12 +210,16 @@ makeKoreLogger exeName startTime timestampSwitch logActionText =
                         toMicroSecs (diffTimeSpec startTime entryTime)
         toMicroSecs = (`div` 1000) . toNanoSecs
     exeName' = Pretty.pretty exeName <> Pretty.colon
-    prettyActualEntry timestamp ActualEntry{actualEntry, entryContext} =
-        (Pretty.vsep . concat)
-            [ [header]
-            , indent <$> context'
-            , indent <$> [longDoc actualEntry]
-            ]
+    prettyActualEntry timestamp ActualEntry{actualEntry, entryContext}
+        | OneLine <- koreLogFormat =
+            fold $
+                catMaybes [timestamp, Just " ", oneLineDoc actualEntry]
```

Comment:
Is it possible to use `Pretty` for formatting here?

Reply:
Since we need everything on one line, we don't need any of the functions in `Pretty`, and the entries in the list are already `Doc` 's

Reply:
The following change should avoid having to add any spaces manually.

```suggestion
            Pretty.hsep $
                catMaybes [timestamp, oneLineDoc actualEntry]
```

I have a concern about this implementation. What happens if a user tries to one-line log an entry which doesn't have `oneLineDoc` implemented? It will just print the timestamp, right? Also, I think that if the user decides to not print timestamps either, nothing will be printed at all, which would probably be very confusing and hard to debug. Maybe we should default to the non-one-line version when the log entry doesn't have `oneLineDoc`, and document this. What do you think?

Reply:
Yes, it should just print the timestamp. At the moment, I don't think there's a way for the user to not print timestamps so I don't think we'll end up with empty lines.
I don't really like the idea of printing long docs when the user wants the one-line format. That can break whatever script the user has in place that would probably consume one line at a time of our output, so for unimplemented stuff we can maybe just have some one-line dummy text that gets put in.
I think the best solution is to actually implement some one-line format for most entries.

Reply:
Wouldn't missing logs also break the user's tool?

There is an option to turn off timestamps, so the situation where the log entry disappears entirely can happen.

Reply:
After some thought, I think we could default `oneLineDoc` to "One line log not implemented." or something, like you suggested.

Reply:
> > > Could you post an example log output to the PR?
> > > Also, I'm a little worried about the performance data on this, for some reason there looks like there's some slow-down on many of the tests.
> > > @ana-pantilie
> > 
> > 
> > ```
> > [8577979] /home/mirceas/work/RV/kore/test/imp/imp.k:35:20-35:67
> > [8587024] /home/mirceas/work/RV/kore/test/imp/imp.k:34:20-34:91
> > [8601394] /home/mirceas/work/RV/kore/test/imp/imp.k:112:8-112:60
> > [8606986] /home/mirceas/work/RV/kore/test/imp/imp.k:34:20-34:91
> > [8616313] /home/mirceas/work/RV/kore/test/imp/imp.k:132:8-132:31
> > [8622051] /home/mirceas/work/RV/kore/test/imp/imp.k:35:20-35:67
> > [8629372] /home/mirceas/work/RV/kore/test/imp/imp.k:133:8-133:24
> > [8635494] /home/mirceas/work/RV/kore/test/imp/imp.k:42:20-43:123
> > [8641144] /home/mirceas/work/RV/kore/test/imp/imp.k:183:8-183:32
> > [8645213] /home/mirceas/work/RV/kore/test/imp/imp.k:154:8-154:16
> > [8649181] /home/mirceas/work/RV/kore/test/imp/imp.k:174:8-174:35
> > [8654153] /home/mirceas/work/RV/kore/test/imp/imp.k:154:8-154:16
> > [8658238] /home/mirceas/work/RV/kore/test/imp/imp.k:174:8-174:35
> > [8662307] /home/mirceas/work/RV/kore/test/imp/imp.k:174:8-174:35
> > [8666586] /home/mirceas/work/RV/kore/test/imp/imp.k:174:8-174:35
> > [8670592] /home/mirceas/work/RV/kore/test/imp/imp.k:174:8-174:35
> > ```
> 
> I just realized, this is missing the log type information. People would need that, so I think it would be good to print all the information we also print in the header during standard logging.

Well, Everett wanted something as concise as possible but I can add that back. I suppose it wouldn't be too hard to skip it.

Reply:
Please add it back. If anyone else wants to use this feature, it might not be for only this log entry.